homeassistant/components/neato/config_flow.py

```diff
@@ -47,7 +54,17 @@ async def async_step_reauth_confirm(self, user_input: dict | None = None) -> dic
     async def async_oauth_create_entry(self, data: dict) -> dict:
         """Create an entry for the flow. Update an entry if one already exist."""
         current_entries = self._async_current_entries()
-        if current_entries and CONF_TOKEN not in current_entries[0].data:
+        if current_entries:
+            neato_session = self.hass.data[NEATO_DOMAIN][current_entries[0].entry_id]
+            hub = NeatoHub(self.hass, Account(neato_session))
+            data["unique_id"] = await hub.async_update_entry_unique_id(
+                current_entries[0]
+            )
+        if (
+            self.source != SOURCE_REAUTH
```

Comment:
This check seems to be incorrect. Probably invert it. If we are re-authenticating we want to update the config entry and abort. Otherwise it's a flow for a new config entry.

Reply:
that part of the flow is needed only for a new config entry; so is excluded during reauth.

Reply:
No, a new config entry will create a new entry. What other cases are there besides reauth that needs to update an existing entry?

Reply:
I though that would create or update. My bad, code updated.